app/app.js

```diff
@@ -156,6 +157,85 @@ app.get(['/config/:configId/((\\d+)).amp', '/((\\d+)).amp'], (req, res) => {
   });
 });
 
+/**
+ * A Showcase article with a server-side paywall.
+ */
+app.get(
+  [
+    '/config/:configId/showcase-article-with-server-side-paywall',
+    '/showcase-article-with-server-side-paywall',
+  ],
+  async (req, res) => {
+    const article = ARTICLES[0];
+    const setup = getSetup(req);
+    const config = getConfig(req.params.configId);
+
+    // Mock a publisher provided user ID, for demo purposes.
+    const publisherProvidedId =
+      'ppidfromscenic' + Math.floor(Math.random() * 9999999999);
+
+    // Mock a registration timestamp, for demo purposes.
+    const registrationTimestamp = Math.floor(Date.now() / 1000);
+
+    // Prepare a request for a Showcase entitlement.
+    const showcaseEntitlementRequestParams = {
+      'metering': {
+        'clientTypes': [
+          1, // Just pass 1 here
+        ],
+        'owner': 'scenic-2017.appspot.com', // Pass your publicationID here
+        'resource': {
+          'hashedCanonicalUrl':
+            '37d9f8ef6e6a2f3e153959b7c126d73cbc9d1d0549ab2a0b389516122cb960d562cbd4cc7d4c8b1b2e488f073c557645088ea2d60f959f847e8b7e7cee931eaf', // Pass your article's hashed canonical URL here
+        },
+        'state': {
+          'id': publisherProvidedId, // Pass your reader's hashed ID
+          'attributes': [
+            {
+              'name': 'standard_registered_user',
+              'timestamp': registrationTimestamp, // Pass registration timestamp in seconds
+            },
+          ],
+        },
+      },
+    };
+
+    // Encode params as Base64 for URLs.
+    // https://en.wikipedia.org/wiki/Base64#URL_applications
+    const encodedParams = Buffer.from(
+      JSON.stringify(showcaseEntitlementRequestParams)
+    )
```

Comment:
Prefer to use toBase64(JSON.stringify(showcaseEntitlementRequestParams)) here.

Or if you are game, you can add this statement as a new util function to crypto.js. It would be good to standardize encoding functions.

Reply:
Good call! I added a new `toBase64Url` method in `crypto.js`